src/signals/IncidentMap/components/IncidentLayer.tsx

```diff
@@ -71,10 +52,40 @@ const IncidentLayer = () => {
     get(`${configuration.GEOGRAPHY_PUBLIC_ENDPOINT}?${searchParams.toString()}`)
   }, [get, bbox])
 
+  useEffect(() => {
+    if (error) {
+      setMapMessage('Er konden geen meldingen worden opgehaald.')
+      return
+    }
+
+    if (!data?.features || !layerInstance) return
+
+    /* istanbul ignore next */
+    data.features.forEach((feature) => {
+      const latlng = featureToCoordinates(feature.geometry)
+      const { name } = feature.properties.category
+
+      const clusteredMarker = L.marker(latlng, {
+        icon: incidentIcon,
+        alt: name,
+        keyboard: false,
+      })
+
+      layerInstance.addLayer(clusteredMarker)
+    })
+
+    return () => {
+      layerInstance.clearLayers()
+    }
+  }, [setMapMessage, layerInstance, data, error])
+
   return (
     <>
       <span data-testid="incidentLayer" />
```

Comment:
Volgens mij wordt dit alleen gebruikt voor het data-testid. Misschien een beetje onnodig om daar een span voor de renderen. Op de map zit ook een data-testid "map-base" welke je ook kunt gebruiken.

Reply:
prima

Reply:
Dat snap ik niet helemaal. Dit is toch het incidentLater component? Dus als die map-base rendered dan heb je toch het component? Je kunt ook nog dubbel checken of MapMessage is gerenderd. (misschien sowieso wel goed aangezien die conditioneel wordt gerenderd)

Reply:
er is een IncidentMap component en een IncidentLayer component. Als er geen data is zou die Layer component misschien niet moeten renderen. Dat ben ik nu ook nog in een andere PR aan het aanpassen. De data wordt in de volgende PR in de IncidentMap component opgehaald.